### ISSUE TITLE
Weird behavior of `autocmd BufEnter * call ncm2#enable_for_buffer()` after updating plugin

### ISSUE THREAD

**Issue by huyvohcmc** [ncm2/ncm2#50]
### OS

MacOS 10.13.4

### neovim `:version` output

0.3.1

### Minimal vimrc for reproducing the issue

```
Plug 'roxma/nvim-yarp'
Plug 'ncm2/ncm2-tagprefix'
Plug 'scrooloose/nerdtree'

autocmd BufEnter * call ncm2#enable_for_buffer()
```
# reproduce the issue

With the recently commits of ncm2, it seems like `autocmd BufEnter * call ncm2#enable_for_buffer()` cause some quirks in vim quickfix: whenever I open `fzf`, `ag` or perform `:PlugUpdate`, vim window doesn't show the result until I hit some keys to get it to update:

![screen shot 2018-08-02 at 1 19 09 pm](https://user-images.githubusercontent.com/17645203/43565898-bad68d1e-9656-11e8-985e-7ae6bda7da10.png)

the status will display `updating...` forever, when I hit `j` or `k` it gets updated like this:

![screen shot 2018-08-02 at 1 19 31 pm](https://user-images.githubusercontent.com/17645203/43565902-befba546-9656-11e8-830d-03679f23c11c.png)

The same happen for `ag` and `fzf` window, they all show blank window:

![screen shot 2018-08-02 at 1 23 11 pm](https://user-images.githubusercontent.com/17645203/43566068-5c75dcba-9657-11e8-8e0f-86d2f2e2529b.png)
![screen shot 2018-08-02 at 1 23 43 pm](https://user-images.githubusercontent.com/17645203/43566072-5f03b18c-9657-11e8-998e-ab12262f19b3.png)

I have to hit a random key to get this:

![screen shot 2018-08-02 at 1 24 06 pm](https://user-images.githubusercontent.com/17645203/43566149-9e779928-9657-11e8-82a5-511b2b74987c.png)

<img width="229" alt="screen shot 2018-08-02 at 1 23 32 pm" src="https://user-images.githubusercontent.com/17645203/43566189-c77e9222-9657-11e8-8448-4c915ac42544.png">

Remove `autocmd BufEnter * call ncm2#enable_for_buffer()` results with no weird errors.

**Comment by clason** [comment]
I get the same behavior also on macOS 10.13.6 and Linux. Bisecting shows that 6f1e1e9ad80a48b6fa67fa565756518ec95e6670 broke this.

**Comment by huyvohcmc** [comment]
@roxma I still get the bug though, the workaround of @clason seems to work so I guess I'll stick with it.

**Comment by roxma** [comment]
@huyvohcmc 

Could you confirm that this issue is fixed after 1e12686?

**Comment by huyvohcmc** [comment]
@roxma Yes it is, thanks so much.
